### PR TITLE
Fix volume button icon

### PR DESCRIPTION
**Pull request by nswarup14**
**Issue**
1. The Volume button on the top of the activity failed to load the 'mute' icon on clicking.

**Changes**
1. `activity.py`

**Features**
On clicking the volume button now, the 'mute icon is loaded' and the sound is disabled as well.

### CODE REVIEW

**Comment by quozl** [comment]
Thanks.  Does it work for you?  When I try the activity at 22f6848, I get _IOError: [Errno 13] Permission denied score.pkl_ and the PyGame canvas is not drawn.  Only widgets that work are the activity menu, mute button or the stop button.

**Comment by nswarup14** [comment]
@quozl @chimosky
I have fixed the game start issue by changing the file path where the file is created. Kindly review and test.

Tested on Sugar desktop v0.114 and Ubuntu 18.04
The game starts with no trouble.
Thanks!

File: main.py [hunk 1]
```diff
@@ -290,16 +291,17 @@ def make(self):
         fruitscore = 0
         score = 0
         
-        if os.path.exists("score.pkl")==False:
-            open('score.pkl','w+')
+        score_path = get_activity_root() + '/instance' + '/score.pkl'
+        if os.path.exists(score_path) is False:
```
Comment (rhl-bthr) [inline main.py]: `if not os.path.exists(score_path)` would be better

File: main.py [hunk 2]
```diff
@@ -290,16 +291,17 @@ def make(self):
         fruitscore = 0
         score = 0
         
-        if os.path.exists("score.pkl")==False:
-            open('score.pkl','w+')
+        score_path = get_activity_root() + '/instance' + '/score.pkl'
```
Comment (rhl-bthr) [inline main.py]: Please use `os.path.join` to join file paths

**Review by rhl-bthr — CHANGES_REQUESTED** [review]
I haven't looked into the issue yet, but `'/instance' + '/score.pkl'` doesn't look like the best place to add the score file

File: main.py (hunk 1 above)
Comment (nswarup14) [reply main.py]: Fixed it @pro-panda

File: main.py (hunk 2 above)
Comment (nswarup14) [reply main.py]: Fixed it @pro-panda

**Comment by quozl** [comment]
It doesn't play.  When activity is started, computer fan speed increases.  When I press PLAY, activity locks up, and computer fan speed increases more.  Have you tested this?  I had a brief look.  It does not follow sugargame sample activity design properly.

Looking at your patch, there's a lot of copy and paste coding.  It is an opportunity to factorise score file access into a separate module; why did you not notice this?  :grin:

File path should be _data_ rather than _instance_.  Many activities get this wrong.  See these pages;

* [Low-Level Activity API - Writable Directories](https://wiki.sugarlabs.org/go/Development_Team/Low-level_Activity_API#Writable_Directories),
* [Development Team - Almanac - How do I get the file path where I can write files?](https://wiki.sugarlabs.org/go/Development_Team/Almanac/sugar.activity.activity#How_do_I_get_the_file_path_where_I_can_write_files_programmatically.3F)

**Comment by nswarup14** [comment]
@quozl 
I tested it in two places. Ubuntu 18.04, via the GNOME terminal and Sugar desktop.
Both the places I'm able to run the activity. I shall attach screenshot as well.
Screenshot from 2019-08-22 07-17-25
Screenshot from 2019-08-22 07-17-35
Screenshot from 2019-08-22 07-17-41

> Looking at your patch, there's a lot of copy and paste coding. It is an opportunity to factorise score file access into a separate module; why did you not notice this?

I noticed actually, but felt it was not that resource intensive, hence it's alright even if I have multiple instances of the score file at different places. I shall refactor.

**Comment by quozl** [comment]
Thanks.  I've investigated.  The performance and heating problem is caused by the choice of 60 frames per second animation and that the entire display surface is updated on each frame.  It is more common to update only the changed parts of the frame.  The Sugargame test activity shows how this is done.  As it was before your work began on the activity, I'll open an issue for it.

Tested eb093be.

**Comment by chimosky** [comment]
Tested eb093be, comments.

It'll be great if the background of `GAME OVER!` extends to the last part of the word.

**Comment by quozl** [comment]
Thanks.  Reviewed.  Will merge and raise feedback as issues.

**Comment by quozl** [comment]
@nswarup14, for your interest, this pull request was difficult to complete because the activity was not mature to begin with.  Many problems existed before your work began, and this made our testing difficult.

**Comment by nswarup14** [comment]
@quozl 
Yes, I understand.
I shall work on the other issues that have been opened up now.

**Pull request merged**